src/main/java/eu/dety/burp/joseph/utilities/Converter.java

```diff
@@ -244,4 +257,124 @@ public static RSAPublicKey getRsaPublicKeyByPemString(String pemInput) {
+            return null;
+        }
+        String kty;
+        ECPublicKey result = null;
+        JSONObject jsonInput = (JSONObject) input;
+        if (jsonInput.containsKey("kty")) {
+            kty = jsonInput.get("kty").toString();
+            if (kty.equals("EC")) {
+                result = buildECPublicKeyByJwk(jsonInput);
+            }
+        } else if (jsonInput.containsKey("epk")) {
+            JSONObject innerJsonArray = (JSONObject) jsonInput.get("epk");
+            kty = (String) innerJsonArray.get("kty");
+            if (kty.equals("EC"))
+                result = buildECPublicKeyByJwk(innerJsonArray);
+        } else {
+            loggerInstance.log(Converter.class, "JSONObject does not contain kty.", Logger.LogLevel.ERROR);
+        }
+        return result;
+    }
+
+    /**
+     * Build EC {@link ECPublicKey} from EC JWK JSON object
+     *
+     * @param input
+     *            EC JSON Web Key {@link JSONObject}
+     * @return {@link ECPublicKey} or null
+     */
+    private static ECPublicKey buildECPublicKeyByJwk(JSONObject input) {
+        if (Security.getProvider(BouncyCastleProvider.PROVIDER_NAME) == null) {
+            Security.addProvider(new BouncyCastleProvider());
+        }
+        try {
+            String crv = input.get("crv").toString();
+            BigInteger x = Base64.decodeInteger(input.get("x").toString().getBytes());
+            BigInteger y = Base64.decodeInteger(input.get("y").toString().getBytes());
+            KeyFactory keyFactory = KeyFactory.getInstance("EC", "BC");
+            ECParameterSpec ecParameterSpec = ECNamedCurveTable.getParameterSpec(crv);
+            ECPoint ecPoint;
+
+            switch (crv) {
+                case "P-256": {
+                    ECFieldElement fex = new SecP256R1FieldElement(x);
+                    ECFieldElement fey = new SecP256R1FieldElement(y);
+                    ecPoint = new SecP256R1Point(ecParameterSpec.getCurve(), fex, fey);
+                    break;
+                }
+                case "P-384": {
+                    ECFieldElement fex = new SecP384R1FieldElement(x);
+                    ECFieldElement fey = new SecP384R1FieldElement(y);
+                    ecPoint = new SecP384R1Point(ecParameterSpec.getCurve(), fex, fey);
+                    break;
+                }
+                case "P-521": {
+                    ECFieldElement fex = new SecP521R1FieldElement(x);
+                    ECFieldElement fey = new SecP521R1FieldElement(y);
+                    ecPoint = new SecP521R1Point(ecParameterSpec.getCurve(), fex, fey);
+                    break;
+                }
+                default:
+                    return null;
+            }
+
+            ECPublicKeySpec ecPublicKeySpec = new ECPublicKeySpec(ecPoint, ecParameterSpec);
+            PublicKey result = keyFactory.generatePublic(ecPublicKeySpec);
+            return (ECPublicKey) result;
+        } catch (InvalidKeySpecException e) {
```

Comment:
You can use multi catch, the error type is not so important to log...you will see it in the e.getMessage()